engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/AnyToTest.java

```diff
@@ -0,0 +1,77 @@
+package org.enso.interpreter.test;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import org.enso.interpreter.runtime.data.EnsoMultiValue;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.test.utils.ContextUtils;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class AnyToTest {
+  private static Context ctx;
+
+  private static final ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+  @BeforeClass
+  public static void initCtx() {
+    ctx = ContextUtils.createDefaultContext(out);
+  }
+
+  @AfterClass
+  public static void disposeCtx() {
+    ctx.close();
+    ctx = null;
+  }
+
+  @Before
+  public void resetOutput() {
+    out.reset();
+  }
+
+  private String getStdOut() {
+    return out.toString(StandardCharsets.UTF_8);
+  }
+
+  @Test
+  public void multiValueToInteger() throws Exception {
+    var ensoCtx = ContextUtils.leakContext(ctx);
+    var types =
+        new Type[] {ensoCtx.getBuiltins().number().getInteger(), ensoCtx.getBuiltins().text()};
+    var code =
+        """
+    from Standard.Base import all
+
+    private eq a b = a == b
+
+    conv style v = case style of
+        0 -> v.to Integer
+        1 -> v:Integer
+        2 -> v.to Text
+        3 -> v:Text
+        99 -> eq
+
+    """;
+    var conv =
+        ContextUtils.evalModule(ctx, Source.newBuilder("enso", code, "conv.enso").build(), "conv");
+    var both = EnsoMultiValue.create(types, types.length, new Object[] {2L, Text.create("Two")});
+    var eq =
+        ContextUtils.executeInContext(
+            ctx,
+            () -> {
+              var bothValue = ctx.asValue(both);
+              var asIntegerTo = conv.execute(0, bothValue);
+              var asIntegerCast = conv.execute(1, bothValue);
```

Comment:
Seems like you forgot to test style 2 and 3, i.e.:
```java
              var asTextTo = conv.execute(2, bothValue);
              var asTextCast = conv.execute(3, bothValue);
assertTrue(equals.execute(asTextTo, asTextCast).asBoolean());
```

Reply:
Good catch, not only it is not called, but it is also broken. These two new tests are failing: d30da41